### PR TITLE
Work on com_finder (both admin and site)

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
Best practices, some optimization and cleaning up.

### Testing Instructions
This should need only code review

### CODE REVIEW

File: administrator/components/com_finder/views/filters/tmpl/default.php [hunk 1]
```diff
@@ -88,9 +88,10 @@
...
 			<tbody>
 				<?php
-				foreach ($this->items as $i => $item) :
 				$canCreate  = $user->authorise('core.create',     'com_finder');
 				$canEdit    = $user->authorise('core.edit',       'com_finder');
+
+				foreach ($this->items as $i => $item) :
 				$canCheckin = $user->authorise('core.manage',     'com_checkin') || $item->checked_out == $user->get('id') || $item->checked_out == 0;
 				$canChange  = $user->authorise('core.edit.state', 'com_finder') && $canCheckin;
 				?>
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/filters/tmpl/default.php]: you could also move `$user->authorise('core.manage',     'com_checkin')` and `$user->authorise('core.edit.state', 'com_finder')` to outside the for since the `$user` does not change inside the for.

File: administrator/components/com_finder/views/filters/view.html.php [hunk 2]
```diff
@@ -20,34 +20,44 @@ class FinderViewFilters extends JViewLegacy
...
+	 *
+	 * @since  2.5
 	 */
 	protected $state;
 
 	/**
 	 * The total number of items
 	 *
-	 * @var  object
+	 * @var  int
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/filters/view.html.php]: integer

File: administrator/components/com_finder/views/filter/view.html.php [hunk 3]
```diff
@@ -20,31 +20,48 @@ class FinderViewFilter extends JViewLegacy
...
+	 * @var  mixed
+	 *
+	 * @since  2.5
 	 */
 	protected $state;
 
 	/**
+	 * The total indexed items
+	 *
+	 * @var  int
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/filter/view.html.php]: integer

File: administrator/components/com_finder/views/index/view.html.php [hunk 4]
```diff
@@ -22,41 +22,53 @@ class FinderViewIndex extends JViewLegacy
...
+	 *
+	 * @since  2.5
 	 */
 	protected $state;
 
 	/**
 	 * The total number of items
 	 *
-	 * @var  object
+	 * @var  int
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/index/view.html.php]: integer

File: administrator/components/com_finder/views/filters/tmpl/default.php (hunk 1 above)
Comment (frankmayer) [inline administrator/components/com_finder/views/filters/tmpl/default.php]: Actually I can't, because the second depends on the first, which may have different results because of the `$item` of the loop. Or am I getting this wrong?

But I could  move the `user->get` outside the loop.

**Comment by andrepereiradasilva** [comment]
@frankmayer when you change tabs identation is more hard for testers to make a code review.

I normally append `?w=1` (ignore whitespaces changes) to the "Files changed" tab URL (Ex: `https://github.com/joomla/joomla-cms/pull/12254/files?w=1`) in those cases to be more easy to spot the differences, but i think not all people know this "trick".

So my sugestion is don't chnange code identation in this PRs

**Comment by frankmayer** [comment]
@andrepereiradasilva As far as I can tell changed indentation on purpose was in one file only.
The other changes of indentation was the removal of nested ifs, resulting in the code of that group being moved to the left.
btw. nice parameter hint

File: administrator/components/com_finder/controllers/indexer.json.php [hunk 5]
```diff
@@ -151,6 +151,8 @@ public function batch()
 
 		// Get the HTML document.
 		$html = JDocument::getInstance('html', $attributes);
+
+		// Todo: Why is this document fetched and immediately overwritten?
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/controllers/indexer.json.php]: good question ...

File: administrator/components/com_finder/controllers/indexer.json.php [hunk 6]
```diff
@@ -164,6 +166,8 @@ public function batch()
 
 		// Swap the app.
 		$app = JFactory::getApplication();
+
+		// Todo: Why is the app fetched and immediately overwritten?
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/controllers/indexer.json.php]: good question ...

File: administrator/components/com_finder/views/filter/view.html.php [hunk 7]
```diff
@@ -20,31 +20,48 @@ class FinderViewFilter extends JViewLegacy
 	 * The filter object
 	 *
 	 * @var  FinderTableFilter
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/filter/view.html.php]: this (and below) were actually added in 3.6.2 see https://github.com/frankmayer/joomla-cms/commit/20c18465afd26df6d7f2c7b7f46124104edfc70f#diff-9d1414e13ac35d173505ccd27d5fc1f5R20

File: administrator/components/com_finder/views/filter/view.html.php [hunk 8]
```diff
@@ -20,31 +20,48 @@ class FinderViewFilter extends JViewLegacy
...
 	protected $state;
 
 	/**
+	 * The total indexed items
+	 *
+	 * @var  integer
+	 *
+	 * @since  2.5
+	 */
+	protected $total;
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/filter/view.html.php]: `__DEPLOY_VERSION__`

File: administrator/components/com_finder/views/filters/tmpl/default.php [hunk 9]
```diff
@@ -88,43 +88,47 @@
...
-					<td class="nowrap hidden-phone">
-						<?php echo $item->map_count; ?>
-					</td>
-					<td class="hidden-phone">
-						<?php echo (int) $item->filter_id; ?>
-					</td>
-				</tr>
+					$canCheckIn = $userAuthoriseCoreManage || $item->checked_out == $userId || $item->checked_out == 0;
+					$canChange = $userAuthoriseCoreEditState && $canCheckIn;
+					$escapedTitle = $this->escape($item->title);
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/filters/tmpl/default.php]: cs: please ident the equal signs

File: administrator/components/com_finder/views/filters/view.html.php [hunk 10]
```diff
@@ -20,34 +20,44 @@ class FinderViewFilters extends JViewLegacy
 	 * An array of items
 	 *
 	 * @var  array
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/filters/view.html.php]: this (and below) was actually added in 3.6.1 see https://github.com/joomla/joomla-cms/commit/20c18465afd26df6d7f2c7b7f46124104edfc70f

File: administrator/components/com_finder/views/index/view.html.php [hunk 11]
```diff
@@ -22,41 +22,53 @@ class FinderViewIndex extends JViewLegacy
 	 * An array of items
 	 *
 	 * @var  array
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/index/view.html.php]: this (and below) was actually added in 3.6.1 see https://github.com/joomla/joomla-cms/commit/20c18465afd26df6d7f2c7b7f46124104edfc70f

File: administrator/components/com_finder/views/maps/view.html.php [hunk 12]
```diff
@@ -22,34 +22,44 @@ class FinderViewMaps extends JViewLegacy
 	 * An array of items
 	 *
 	 * @var  array
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/maps/view.html.php]: this (and below) was actually added in 3.6.1 see https://github.com/joomla/joomla-cms/commit/20c18465afd26df6d7f2c7b7f46124104edfc70f

File: administrator/components/com_finder/views/statistics/view.html.php [hunk 13]
```diff
@@ -20,6 +20,8 @@ class FinderViewStatistics extends JViewLegacy
 	 * The index statistics
 	 *
 	 * @var  JObject
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/statistics/view.html.php]: this (and below) was actually added in 3.6.1 see https://github.com/joomla/joomla-cms/commit/20c18465afd26df6d7f2c7b7f46124104edfc70f

File: components/com_finder/views/search/view.html.php [hunk 14]
```diff
@@ -24,6 +24,12 @@ class FinderViewSearch extends JViewLegacy
 
 	protected $user;
 
+	protected $results;
+
+	protected $total;
+
+	protected $pagination;
```
Comment (andrepereiradasilva) [inline components/com_finder/views/search/view.html.php]: please add doc blocks

Comment (frankmayer) [inline components/com_finder/views/search/view.html.php]: I removed one protected property, as it seemed not be used.

File: administrator/components/com_finder/views/filter/view.html.php [hunk 15]
```diff
@@ -20,31 +20,48 @@ class FinderViewFilter extends JViewLegacy
...
+	 * @since  3.6.2
 	 */
 	protected $filter;
 
 	/**
 	 * The JForm object
 	 *
 	 * @var  JForm
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/filter/view.html.php]: 3.6.2

File: administrator/components/com_finder/views/filter/view.html.php [hunk 16]
```diff
@@ -20,31 +20,48 @@ class FinderViewFilter extends JViewLegacy
...
 	 */
 	protected $form;
 
 	/**
 	 * The active item
 	 *
-	 * @var  object
+	 * @var  JObject|boolean
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/filter/view.html.php]: 3.6.2

File: administrator/components/com_finder/views/filter/view.html.php [hunk 17]
```diff
@@ -20,31 +20,48 @@ class FinderViewFilter extends JViewLegacy
...
 	 */
 	protected $item;
 
 	/**
 	 * The model state
 	 *
-	 * @var  object
+	 * @var  mixed
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/filter/view.html.php]: 3.6.2

File: administrator/components/com_finder/views/index/view.html.php [hunk 18]
```diff
@@ -22,41 +22,53 @@ class FinderViewIndex extends JViewLegacy
...
 	 */
 	protected $items;
 
 	/**
 	 * The pagination object
 	 *
 	 * @var  JPagination
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/index/view.html.php]: 3.6.1

File: administrator/components/com_finder/views/index/view.html.php [hunk 19]
```diff
@@ -22,41 +22,53 @@ class FinderViewIndex extends JViewLegacy
...
 	 */
 	protected $pagination;
 
 	/**
 	 * The state of core Smart Search plugins
 	 *
 	 * @var  array
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/index/view.html.php]: 3.6.1

File: administrator/components/com_finder/views/index/view.html.php [hunk 20]
```diff
@@ -22,41 +22,53 @@ class FinderViewIndex extends JViewLegacy
...
 	 */
 	protected $pluginState;
 
 	/**
 	 * The HTML markup for the sidebar
 	 *
 	 * @var  string
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/index/view.html.php]: 3.6.1

File: administrator/components/com_finder/views/index/view.html.php [hunk 21]
```diff
@@ -22,41 +22,53 @@ class FinderViewIndex extends JViewLegacy
...
 	 */
 	protected $sidebar;
 
 	/**
 	 * The model state
 	 *
-	 * @var  object
+	 * @var  mixed
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/index/view.html.php]: 3.6.1

File: administrator/components/com_finder/views/index/view.html.php [hunk 22]
```diff
@@ -22,41 +22,53 @@ class FinderViewIndex extends JViewLegacy
...
 	 */
 	protected $state;
 
 	/**
 	 * The total number of items
 	 *
-	 * @var  object
+	 * @var  integer
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/index/view.html.php]: 3.6.1

File: administrator/components/com_finder/views/maps/view.html.php [hunk 23]
```diff
@@ -22,34 +22,44 @@ class FinderViewMaps extends JViewLegacy
...
 	 */
 	protected $items;
 
 	/**
 	 * The pagination object
 	 *
 	 * @var  JPagination
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/maps/view.html.php]: 3.6.1

File: administrator/components/com_finder/views/maps/view.html.php [hunk 24]
```diff
@@ -22,34 +22,44 @@ class FinderViewMaps extends JViewLegacy
...
 	 */
 	protected $pagination;
 
 	/**
 	 * The HTML markup for the sidebar
 	 *
 	 * @var  string
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/maps/view.html.php]: 3.6.1

File: administrator/components/com_finder/views/maps/view.html.php [hunk 25]
```diff
@@ -22,34 +22,44 @@ class FinderViewMaps extends JViewLegacy
...
 	 */
 	protected $sidebar;
 
 	/**
 	 * The model state
 	 *
 	 * @var  object
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/maps/view.html.php]: 3.6.1

File: administrator/components/com_finder/views/maps/view.html.php [hunk 26]
```diff
@@ -22,34 +22,44 @@ class FinderViewMaps extends JViewLegacy
...
 	 */
 	protected $state;
 
 	/**
 	 * The total number of items
 	 *
 	 * @var  object
+	 *
+	 * @since  2.5
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/views/maps/view.html.php]: 3.6.1

File: components/com_finder/views/search/view.html.php [hunk 27]
```diff
@@ -16,13 +16,53 @@
...
+     * @since  3.6.1
+     */
+	protected $results;
+
+    /**
+     * The total number of items
+     *
+     * @var    integer
+     *
+     * @since  3.6.1
```
Comment (andrepereiradasilva) [inline components/com_finder/views/search/view.html.php]: `__DEPLOY_VERSION__`

File: components/com_finder/views/search/view.html.php [hunk 28]
```diff
@@ -16,13 +16,53 @@
...
+     */
 	protected $state;
 
-	protected $user;
+    /**
+     * An array of results
+     *
+     * @var    array
+     *
+     * @since  3.6.1
```
Comment (andrepereiradasilva) [inline components/com_finder/views/search/view.html.php]: `__DEPLOY_VERSION__`

File: components/com_finder/views/search/view.html.php [hunk 29]
```diff
@@ -16,13 +16,53 @@
...
 	protected $params;
 
+    /**
+     * The model state
+     *
+     * @var  object
+     */
 	protected $state;
 
-	protected $user;
```
Comment (andrepereiradasilva) [inline components/com_finder/views/search/view.html.php]: why is this removed? is B/C to remove a protected property?

Comment (frankmayer) [inline components/com_finder/views/search/view.html.php]: Yes, it could be. I was overeagerly cleaning up and noticed, it wasn't used in the context of this file or related tmpl. Will revert.

File: components/com_finder/views/search/view.html.php [hunk 30]
```diff
@@ -16,14 +16,56 @@
  */
 class FinderViewSearch extends JViewLegacy
 {
+    /**
```
Comment (andrepereiradasilva) [inline components/com_finder/views/search/view.html.php]: cs: please ad tabs for indentation (instead of spaces) in this doc blocks (like the other doc blocks in other files)

Comment (frankmayer) [inline components/com_finder/views/search/view.html.php]: Sh*t! somehow a commit from another branch got into this.  pls ignore..  fixing now

File: components/com_finder/views/search/view.html.php [hunk 31]
```diff
@@ -164,23 +208,23 @@ protected function getLayoutFile($layout = null)
...
+		$exists     = JPath::find($this->_path['template'], $filetofind);
 
 		return ($exists ? $layout : 'result');
 	}
 
 	/**
 	 * Prepares the document
 	 *
-	 * @param   FinderIndexerQuery  $query  The search query
+	 * @param   FinderIndexerQuery $query The search query
```
Comment (andrepereiradasilva) [inline components/com_finder/views/search/view.html.php]: cs: revert this change

File: components/com_finder/views/search/view.html.php [hunk 32]
```diff
@@ -150,7 +194,7 @@ protected function getFields()
 	/**
 	 * Method to get the layout file for a search result object.
 	 *
-	 * @param   string  $layout  The layout file to check. [optional]
+	 * @param   string $layout The layout file to check. [optional]
```
Comment (andrepereiradasilva) [inline components/com_finder/views/search/view.html.php]: cs: revert this change

File: components/com_finder/views/search/view.html.php [hunk 33]
```diff
@@ -9,33 +9,76 @@
...
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $pagination;
+
+	/**
 	 * Method to display the view.
 	 *
-	 * @param   string  $tpl  A template file to load. [optional]
+	 * @param   string $tpl A template file to load. [optional]
```
Comment (andrepereiradasilva) [inline components/com_finder/views/search/view.html.php]: cs: revert this change

File: components/com_finder/views/search/view.html.php (hunk 31 above)
Comment (frankmayer) [inline components/com_finder/views/search/view.html.php]: Already done... must check the autoformatter for the project in my ide

Comment (frankmayer) [inline components/com_finder/views/search/view.html.php]: ... wasn't sure of the spaces, so I risked it. my code sniffer didn't complain though. Must check what's wrong there...

**Review by andrepereiradasilva — APPROVED** [review]

**Comment by frankmayer** [comment]
I think the drone failure is a hickup. Could someone with permissions restart the check?

File: administrator/components/com_finder/models/maps.php [hunk 34]
```diff
@@ -345,16 +345,13 @@ public function publish(&$pks, $value = 1)
 		{
 			$table->reset();
 
-			if ($table->load($pk))
+			if ($table->load($pk) && !$this->canEditState($table))
```
Comment (Quy) [inline administrator/components/com_finder/models/maps.php]: Dedent block below.

File: administrator/components/com_finder/tables/filter.php [hunk 35]
```diff
@@ -168,12 +168,12 @@ public function publish($pks = null, $state = 1, $userId = 0)
 		}
 
 		// If checkin is supported and all rows were adjusted, check them in.
-		if ($checkin && (count($pks) == $this->_db->getAffectedRows()))
+		if ($checkin && (count($pks) === $this->_db->getAffectedRows()))
```
Comment (Quy) [inline administrator/components/com_finder/tables/filter.php]: Remove outside parentheses `(count($pks) === $this->_db->getAffectedRows())`

File: administrator/components/com_finder/views/index/tmpl/default.php [hunk 36]
```diff
@@ -118,7 +118,7 @@
 						<?php echo JHtml::_('date', $item->indexdate, JText::_('DATE_FORMAT_LC4')); ?>
 					</td>
 					<td class="center hidden-phone hidden-tablet">
-						<?php if (intval($item->publish_start_date) or intval($item->publish_end_date) or intval($item->start_date) or intval($item->end_date)) : ?>
+						<?php if ((int) $item->publish_start_date or (int) $item->publish_end_date or (int) $item->start_date or (int) $item->end_date) : ?>
```
Comment (Quy) [inline administrator/components/com_finder/views/index/tmpl/default.php]: Change `or` to `||`.

File: administrator/components/com_finder/views/statistics/tmpl/default.php [hunk 37]
```diff
@@ -34,7 +34,7 @@
 						<?php
 						$lang_key    = 'PLG_FINDER_STATISTICS_' . str_replace(' ', '_', $type->type_title);
 						$lang_string = JText::_($lang_key);
-						echo ($lang_string == $lang_key) ? $type->type_title : $lang_string;
+						echo ($lang_string === $lang_key) ? $type->type_title : $lang_string;
```
Comment (Quy) [inline administrator/components/com_finder/views/statistics/tmpl/default.php]: Remove parentheses.

**Comment by frankmayer** [comment]
Thanks for reviewing @Quy

**Comment by franz-wohlkoenig** [comment]
@frankmayer is this PR to be tested by Code review?

File: components/com_finder/models/search.php [hunk 38]
```diff
@@ -425,7 +425,7 @@ protected function getResultsTotal()
 					$temp = $this->_db->loadObjectList();
 
 					// Set the more flag to true if any of the sets equal the limit.
-					$more = (count($temp) === $limit) ? true : false;
+					$more = (count($temp) === $limit);
```
Comment (Quy) [inline components/com_finder/models/search.php]: Sorry found more. Remove parentheses. Do the same 3 more times below.

**Comment by Quy** [comment]
@franz-wohlkoenig yes code review.

**Comment by franz-wohlkoenig** [comment]
@Quy thanks for Info. @frankmayer can you please wrote Test Instructions "Code review" in first Comment for willing Tester?

**Comment by frankmayer** [comment]
@Quy, @franz-wohlkoenig : Done and Done. Thanks

**Comment by frankmayer** [comment]
Conflicts resolved. Please test 
Code review only should do.

File: administrator/components/com_finder/models/filter.php [hunk 39]
```diff
@@ -168,8 +168,7 @@ public function getTotal()
 			->select('MAX(link_id)')
 			->from('#__finder_links');
 		$db->setQuery($query);
-		$total = $db->loadResult();
 
-		return $total;
+		return $db->loadResult();
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/models/filter.php]: small improv: you could use `return $db->setQuery($query)->loadResult();` and remove the above line

File: administrator/components/com_finder/controllers/indexer.json.php [hunk 40]
```diff
@@ -165,6 +165,8 @@ public function batch()
 
 		// Get the HTML document.
 		$html = JDocument::getInstance('html', $attributes);
+
+		// Todo: Why is this document fetched and immediately overwritten?
 		$doc = JFactory::getDocument();
```
Comment (andrepereiradasilva) [inline administrator/components/com_finder/controllers/indexer.json.php]: i think you can safely remove this line

File: components/com_finder/views/search/view.html.php [hunk 41]
```diff
@@ -9,22 +9,65 @@
 
 defined('_JEXEC') or die;
 
+
```
Comment (andrepereiradasilva) [inline components/com_finder/views/search/view.html.php]: please remove this empty line

**Review by andrepereiradasilva — COMMENTED** [review]
just some minor comments
rest seems fine on code review

File: administrator/components/com_finder/models/filter.php (hunk 39 above)
Comment (frankmayer) [inline administrator/components/com_finder/models/filter.php]: Totally missed your comments here. Commit coming up.

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 222ab4c7458117f098c268a94a4324a80830df14<br /><br />code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12254">issues.joomla.org/tracker/joomla-cms/12254</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 222ab4c7458117f098c268a94a4324a80830df14<br /><br />Code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12254">issues.joomla.org/tracker/joomla-cms/12254</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.
